src/parseData.js

```diff
@@ -1,10 +1,15 @@
 /**
  * The resulting parse tree nodes of the parse tree.
  */
-function ParseNode(type, value, mode) {
+function ParseNode(type, value, mode, firstToken, lastToken) {
     this.type = type;
     this.value = value;
     this.mode = mode;
+    if (firstToken) {
+        this.lexer = firstToken.lexer;
+        this.start = firstToken.start;
+        this.end = (lastToken || firstToken).end;
+    }
```

Comment:
I'm confused by this.  Why does a `ParseNode` need a reference to `firstToken`'s lexer?